### DISCUSSION TITLE
How to avoid ERR_CONNECTION_TIMED_OUT page?

### DISCUSSION THREAD

**Discussion by damil42** [keycloak/keycloak#29655] [Q&A]
I use keycloak-js. When I call keycloak.init, it's navigation to the "This site can't be reached." (ERR_CONNECTION_TIMED_OUT ) page of the browser, when the server is not available (connection timeout).

```js
keycloak.init({
 flow: 'standard',
 onLoad: 'check-sso',
 checkLoginIframe: false,
});
```

Like this:

<img src="https://github.com/keycloak/keycloak/assets/126908176/d3d45152-5209-4117-a372-427471851183" width="250" />

How to prevent this behavior and catch this error to let the dev decide if and how to show this error?

Try-catch, promise-catch and onAuthError do not work in my tests. With the current behavior, you can't use the website anymore. Even if the SSO is optional. I want check-sso on load. But don't block the website on timeout.

I could use `fetch` to test the connection before keycloak `init`. But I would avoid this workaround and find a better way.

**Comment by LyraHarring** [comment]
The error [ERR_CONNECTION_TIMED_OUT](https://certera.com/kb/how-to-fix-the-err-connection-timed-out-error/) indicates that you need to focus does not indicate a dangerous problem or an infected file that can harm your system. Instead, it denotes that the system could not connect successfully with the target server. It does not pop up because of a problem on the user’s end. The error in question can also originate due to issues on the server side, mainly if the website is experiencing high traffic or resource constraints, which is mostly the case with shared hosting platforms. The reasons for the error to be occured which are as follows:-

1) Invalid URL
2) Slow Internet connection
3) Firewall settings
4) Server delay or error

There are many different ways to fix the error that you need to understand in detail.

**Comment by Eden-Allen** [comment]
One of the most annoying errors is when your internet connection times out, and you can’t get to a certain page. [ERR_CONNECTION_TIMED_OUT](https://cheapsslweb.com/blog/fix-err_connection_timed_out-error-in-chrome/) will show up on the screen with the message “The page is not available.”

This error usually happens when the internet connection isn’t working right, and your WordPress website is trying to load more data from the database. When a website or page takes too long to load, or the connection times out, a browser like Chrome gives you three ways to fix the problem.
 
1) Checking the connection to the network
2) Run the Windows Network Diagnostics
3) Assess the firewall and proxy
 
There are several steps involved in solving the error, which you need to understand in detail. Hope it helps!
